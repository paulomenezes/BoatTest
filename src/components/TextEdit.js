import React from 'React';
import { TextInput, View } from 'react-native';
import styles from './TextEdit.styles';

const TextEdit = props => {
  const { input, meta, style, ...inputProps } = props;

  const validationStyles =
    meta.touched && !meta.active
      ? !meta.error ? styles.validField : styles.invalidField
      : styles.validField;

  return (
    <View style={styles.container}>
      <View style={[style, styles.inputArea, validationStyles]}>
        <TextInput
          style={styles.input}
          placeholderTextColor="#020202"
          {...inputProps}
          onChangeText={input.onChange}
          onBlur={input.onBlur}
          onFocus={input.onFocus}
          value={input.value}
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
  );
};

export default TextEdit;
