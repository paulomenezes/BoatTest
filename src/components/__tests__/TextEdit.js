import 'react-native';
import React from 'react';
import TextEdit from '../TextEdit.js';

import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

it('renders correctly', () => {
  const props = {
    meta: {
      touched: false,
      active: false,
      error: undefined
    },
    input: {}
  };

  const tree = renderer.create(<TextEdit {...props} />);
  expect(tree).toMatchSnapshot();
});

it('Gray border when not touched', () => {
  const props = {
    meta: {
      touched: false,
      active: true,
      error: undefined
    },
    input: {}
  };

  const renderer = new ShallowRenderer();
  renderer.render(<TextEdit {...props} />);

  const fieldBorder = renderer.getRenderOutput().props.children.props.style[2]
    .borderColor;

  expect(fieldBorder).toBe('#CFCFCF');
});

it('Gray border when is valid', () => {
  const props = {
    meta: {
      touched: true,
      active: false,
      error: undefined
    },
    input: {}
  };

  const renderer = new ShallowRenderer();
  renderer.render(<TextEdit {...props} />);

  const fieldBorder = renderer.getRenderOutput().props.children.props.style[2]
    .borderColor;

  expect(fieldBorder).toBe('#CFCFCF');
});

it('Gray border when invalid', () => {
  const props = {
    meta: {
      touched: true,
      active: false,
      error: true
    },
    input: {}
  };

  const renderer = new ShallowRenderer();
  renderer.render(<TextEdit {...props} />);

  const fieldBorder = renderer.getRenderOutput().props.children.props.style[2]
    .borderColor;

  expect(fieldBorder).toBe('#D75452');
});
