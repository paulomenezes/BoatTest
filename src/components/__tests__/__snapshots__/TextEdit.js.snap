// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 40,
          "shadowColor": "#E6E6E6",
          "shadowOffset": Object {
            "height": 2,
            "width": 2,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
        },
        Object {
          "borderColor": "#CFCFCF",
        },
      ]
    }
  >
    <TextInput
      onBlur={undefined}
      onChangeText={undefined}
      onFocus={undefined}
      placeholderTextColor="#020202"
      style={
        Object {
          "flex": 1,
          "fontFamily": "OpenSans-Regular",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
        }
      }
      value={undefined}
    />
  </View>
</View>
`;
